### PR TITLE
Fix #208 - java-junit: show annotations on PR changed files

### LINKED ISSUE #208: [java-junit] show annotations on PR "Files changed" tab next to code line related to the failure
# Current situation
Currently `show annotations on pull request "Files changed" tab next to code line related to the failure` works nicely when JUnit report is generated for example on NodeJS using jest-junit reporter:
image

but the same feature doesn't work on Java Virutal Machine (Java/Kotlin/Groovy/...)

# Problem
Currently the [code that parses exception stack trace lines](https://github.com/dorny/test-reporter/blob/074fe2cd2777c59e9e376f717091cfe8d86be90e/src/parsers/java-junit/java-junit-parser.ts#L147-L152) fails to parse the line if the line:
* starts with whitespace (maybe there are some frameworks that generate JUnit xml report file where stack trace lines don't start with whitespaces, but that isn't the case for gradle with JUnit - probably dominant behavior)
* when running on Java Runtime Environment version 9 or newer, when application code is loaded with custom ClassLoader (like Spring Boot) or when application source code is in Java 9 module

Example of stack trace line, that isn't parsed:
>    	at app//com.foo.Main.run(Main.java:101)

# Expected behaviour
On JVM (Java/Kotlin/Groovy/...) annotations should be added to pull request "Files changed" tab next to code line related to the failure based on stack trace

### PR DESCRIPTION
**Pull request by atsu85**
Fixes #208.
Resulting screenshot of annotation added to the PR of the project that i used for testing the changes:
image

### CODE REVIEW

File: .nvmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+v18.7.0
```
Comment (atsu85) [inline .nvmrc:1]: FYI: this is for automatic version switching when using [nvm](https://github.com/nvm-sh/nvm) or [fvm](https://github.com/Schniz/fnm#--fast-node-manager-fnm----)

File: .gitignore [hunk 2]
```diff
@@ -100,3 +100,5 @@ lib/**/*
 
 # Project specific
 __tests__/__results__
+
+.idea
```
Comment (atsu85) [inline .gitignore:104]: FYI: for ignoring IntelliJ idea IDE specific files

**Comment by dharmendrasha** [comment]
https://github.com/dorny/test-reporter/actions/runs/3505027534/jobs/5983498492

Showing me this error in the code pipeline

image

**Comment by atsu85** [comment]
Hi @dharmendrasha, 
Please reopen this PR - you came to wrong conclusion!

This error you pointed out was expected, because this PR was created by me and i don't have permissions for this repo (similar situation [described here](https://github.com/actions/labeler/issues/12#issuecomment-525477638)). However if the owner of this repo creates the PR from my commits, then this error shouldn't occur, just like [it didn't occur in my fork using the same commits](https://github.com/atsu85/test-reporter/actions/runs/3562764332/jobs/5984791977)

**Comment by dharmendrasha** [comment]
Sure @atsu85 will do that

**Comment by atsu85** [comment]
@dharmendrasha , by the way [the same issue is even reported even in this project many times](https://github.com/search?q=repo%3Adorny%2Ftest-reporter+%22Resource+not+accessible+by+integration%22&type=issues) and there is even [a PR](https://github.com/dorny/test-reporter/pull/174) (by @hungrybear88) to fix it ([as claimed here](https://github.com/dorny/test-reporter/issues/180#issuecomment-1168503419) by @hilmarf), but currently that PR isn't merged

**Comment by atsu85** [comment]
> Sure @atsu85 will do that

@dharmendrasha, did you forget to reopen this PR? :)

**Review by dorny — APPROVED** [review]

**Comment by dorny** [comment]
Thanks @atsu85 ! PR looks good. I'm merging it now.
It's true the PR's from forked repos will have failed check run due to permission issues.
Shame on me, the issue and workaround are actually described in our own [README](https://github.com/dorny/test-reporter#recommended-setup-for-public-repositories).
Probably we should use that approach for the PR's and test the direct approach some other way.

**Pull request merged**